pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/io/ConnectorUtils.java

```diff
@@ -142,49 +149,83 @@ public static TreeMap<String, Connector> searchForConnectors(String connectorsDi
         Path path = Paths.get(connectorsDirectory).toAbsolutePath();
         log.info("Searching for connectors in {}", path);
 
-        TreeMap<String, Connector> connectors = new TreeMap<>();
         if (!path.toFile().exists()) {
             log.warn("Connectors archive directory not found");
-            return connectors;
+            return new TreeMap<>();
```

Comment:
Collections.emptyMap() ?

Reply:
Collections.emptyMap() doesn't work with TreeMap. Please note this code is invoked at startup only if the concern is repetitive initialization. Interestingly, I couldn't find a java.util or guava API to create an immutable TreeMap.